tests/writer_util.py

```diff
@@ -76,26 +81,26 @@ def _convert_clob_pairs(clob_pairs):
 
 _SIMPLE_SYMBOLS_TEXT=(
     (u'', br"''"),
-    (u'\u0000', br"'\x00'"),
+    (u'\u0000', (br"'\x00'", br"'\0'")),
     (u'4hello', br"'4hello'"),
     (u'hello', br"hello"),
     (u'_hello_world', br"_hello_world"),
     (u'null', br"'null'"),
     (u'hello world', br"'hello world'"),
     (u'hello\u0009\x0a\x0dworld', br"'hello\t\n\rworld'"),
-    (u'hello\aworld', br"'hello\x07world'"),
-    (u'hello\u3000world', br"'hello\u3000world'"), # A full width space.
-    (u'hello\U0001f4a9world', br"'hello\U0001f4a9world'"), # A 'pile of poo' emoji code point.
+    (u'hello\aworld', (br"'hello\x07world'", br"'hello\aworld'")),
+    (u'hello\u3000world', (br"'hello\u3000world'", b"'hello\xe3\x80\x80world'")),  # A full width space.
+    (u'hello\U0001f4a9world', (br"'hello\U0001f4a9world'", b"'hello\xf0\x9f\x92\xa9world'")),  # A 'pile of poo' emoji code point.
```

Comment:
I don't think the `\x` representations are equivalent in this case. `\u` and `\U` indicate  unicode code points, while `\x` is just for raw bytes. Maybe there's a flaw in ion-c's unicode escaping logic?

Reply:
As discussed offline,  I noticed that 'b' prefix treats bytes as a whole here.

I did some experiment that shows unicode representation of `u'\U0001f4a9'` is equivalent to `b'\xf0\x9f\x92\xa9'`
 
```.py
a = u'\U0001f4a9'.encode('utf-8')
b = b'\xf0\x9f\x92\xa9'
print(a)    # 💩
print(b)      # 💩
print(a == b)    # True
```